core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp

```diff
@@ -237,6 +237,25 @@ inline size_t get_shmem_per_sm_prefer_l1(cudaDeviceProp const& properties) {
     return 0;
   }() * 1024;
 }
+
+// Assuming !cudaFuncSetCacheConfig(MyKernel, cudaFuncCachePreferL1) this
+// routine lists the approximate size for 50% cache size based on Cuda occupance
+// calculator spreadsheet.
+inline size_t get_shmem_per_sm_prefer_equal(cudaDeviceProp const& properties) {
+  int const compute_capability = properties.major * 10 + properties.minor;
+  return [compute_capability]() {
+    switch (compute_capability) {
+      case 70:
+      case 80: return 64;
+      case 86: return 96;
+      case 75: return 32;
+      default:
+        Kokkos::Impl::throw_runtime_exception(
+            "Unknown device in cuda block size deduction");
```

Comment:
We need to make sure we do not throw for other valid architectures.
I would list all of them and return `0`

Reply:
Fixed it.